backend/module/chaos/experimentation/xds/xds.go

```diff
@@ -244,11 +304,25 @@ func (c *ecdsCallbacks) OnStreamRequest(streamID int64, req *gcpDiscoveryV3.Disc
 	return nil
 }
 
+func (c *ecdsCallbacks) OnStreamDeltaRequest(streamID int64, req *gcpDiscoveryV3.DeltaDiscoveryRequest) error {
+	c.safeECDSResources.setResourcesForCluster(req.Node.Cluster, req.ResourceNamesSubscribe)
```

Comment:
this is the only line i'm not sure about. there's also `req.ResourceNamesUnsubscribe`. afaict that could be used for optimization but won't cause any problems if left out based on my reading of the other code, in particular `getResourcesForCluster`.

Reply:
Afair, the whole logic around `ecdsResourceMap` is used for a fail-safe mechanism. 
Eg, Let's say we have injected a fault on a cluster but at the time of removal, the cluster is disconnected from the server. Whenever the connection re-establishes, it will remove the fault.